### PR TITLE
refactor(components): standardize labware outline alignment and highlight state

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Fixes issues where the LabwareOutline component was causing visual skew in x-y positions of labware
renders. Adds support to the MoveLabwareOnDeck component for deck configuration through usage of
BaseDeck

# Risk assessment
medium, the `LabwareRender`, `LabwareOutline`, and `StaticLabware` components are widely used

### CODE REVIEW

File: components/src/hardware-sim/RobotCoordinateSpace/RobotCoordinateSpace.tsx [hunk 1]
```diff
@@ -1,13 +1,27 @@
+import styled from 'styled-components'
+import { animated } from '@react-spring/web'
```
Comment (jerader) [inline components/src/hardware-sim/RobotCoordinateSpace/RobotCoordinateSpace.tsx:2]: i didn't realize we had `react-spring` in our dependencies

File: components/src/hardware-sim/Deck/MoveLabwareOnDeck.tsx [hunk 2]
```diff
@@ -13,17 +13,18 @@ import {
...
   Coordinates,
   LabwareDefinition2,
   LabwareLocation,
   RobotType,
   DeckDefinition,
+  DeckConfiguration,
 } from '@opentrons/shared-data'
 import type { StyleProps } from '../../primitives'
 import type { TrashCutoutId } from './FlexTrash'
+import { BaseDeck } from '../BaseDeck'
```
Comment (jerader) [inline components/src/hardware-sim/Deck/MoveLabwareOnDeck.tsx]: import can go higher

**Review by jerader — APPROVED** [review]
did a quick audit of PD, LL, and the app and things are looking good! 🖌️

File: components/src/hardware-sim/Deck/MoveLabwareOnDeck.stories.tsx [hunk 3]
```diff
@@ -14,16 +23,70 @@ const meta: Meta<React.ComponentProps<typeof MoveLabwareOnDeckComponent>> = {
 export default meta
 type Story = StoryObj<React.ComponentProps<typeof MoveLabwareOnDeckComponent>>
 
+export const FLEX_SIMPLEST_DECK_CONFIG: DeckConfiguration = [
```
Comment (brenthagen) [inline components/src/hardware-sim/Deck/MoveLabwareOnDeck.stories.tsx]: not important, but the `export` is causing storybook to attempt to render `FLEX_SIMPLEST_DECK_CONFIG` as a story

**Review by brenthagen — APPROVED** [review]
LGTM !

**Pull request merged**